### DISCUSSION TITLE
Does cloudflare turnstile can't be bypass from seleniumbase for now . the cloudflare turnstile is sucess but error to find element the code is below

### DISCUSSION THREAD

**Discussion by Leeegod** [seleniumbase/SeleniumBase#2496] [Q&A] [answered]
from seleniumbase import SB

def openpage(sb):
    sb.driver.uc_open_with_reconnect(
        "https://visa.vfsglobal.com/are/en/fra/login/#relax", reconnect_time=60,
    )

def verify(sb):
    # sb.driver.uc_switch_to_frame("iframe")
    sb.driver.js_click("span.mark")
    
with SB(uc=True) as sb:
    openpage(sb)
    try:
        verify(sb)
    except Exception:
        openpage(sb)
        verify(sb)
    sb.sleep(2)

**Comment by mdmintz** [comment, marked as answer]
If you need clicks to go undetected, try the UC Mode examples that click undetected:
* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/raw_turnstile.py
* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/raw_form_turnstile.py

(Those checkboxes are located in iframes, so you need to switch into them first, carefully.)

**Reply by Leeegod** [reply]
tried uc mode ist sucessfully get sucess in cloudflare turestile but can't get other element to to send the other value

**Reply by mdmintz** [reply]
There's already a lot for that website on Stack Overflow: https://stackoverflow.com/search?q=https%3A%2F%2Fvisa.vfsglobal.com

**Comment by OpsecGuy** [comment]
Damn so many of you guys have issues with this so let's self-leak my insane method for getting rid of that annoying captcha.
Because by default seleniumbase with UC mode ain't automatically solve/ bypass Turnstaile especially in web forms here is ultimate solution:

```
# @Author: github.com/OpsecGuy
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
# from selenium.common.exceptions import TimeoutException, NoSuchElementException

def cf_manual_solver(self) -> None:
    captcha_frame_regex = re.compile(r'cf-chl-widget-.{3,6}')
    try:
        # Usually it's item with index 0
        matching_elements = self.session.find_elements(by=By.XPATH, value="//*[contains(@id, 'cf-chl-widget-')]")
        # print(f'Matches found: {matching_elements}')
        for element in matching_elements:
            element_id = element.get_attribute("id")
            if captcha_frame_regex.match(element_id) and 'Cloudflare security challenge' in element.accessible_name:
                # print(f"Matched Element ID: {element_id} - {element.accessible_name}")
                cf_captcha_frame = element
                break

        try:
            # Switch to captcha iframe
            WebDriverWait(self.session, 15.0).until(EC.frame_to_be_available_and_switch_to_it(cf_captcha_frame))
            captcha_checkbox = self.session.find_element(by=By.CLASS_NAME, value='ctp-checkbox-label')
            captcha_checkbox.click()
            # Back to default content
            self.session.switch_to.default_content()
        except Exception as err:
            print('Failed to click captcha checkbox!')
            # return None
```

So after you add this to your codebase usage would be like this:
PSEUDO CODE
```
driver = Driver(uc=True)
driver.uc_open_with_reconnect("https://visa.vfsglobal.com/are/en/fra/login#relax", 5.0)
driver.sleep(2.5) # Make sure everything on page is loaded
cf_manual_solver()
...
do your stuff here
```

In 99% of cases code above works. Sometimes you might get 'Captcha solve failure' which can be caused by plenty of things and more research would be required. Also, I noticed that your target changed Turnstaile to ReCaptcha after a few requests in a row so you would need to implement bypass for it as well.

**Reply by Leeegod** [reply]
After the function is call its show the error TypeError: cf_manual_solver() missing 1 required positional argument: 'self'

**Reply by OpsecGuy** [reply]
Remove the "self". I copied it from my old project. It was a method inside of class. Don't copy-paste this blindly 🤔

**Reply by Leeegod** [reply]
matching_elements = self.session.find_elements(by=By.XPATH, value="//*[contains(@id, 'cf-chl-widget-')]")
session can't be found didn''t understand here

**Reply by OpsecGuy** [reply]
self.session is a reference to the driver object.

**Comment by Leeegod** [comment]
Thanks Got it

**Comment by yell0w4x** [comment]
Looks like it's due to it's in shadow root dom 
https://stackoverflow.com/questions/78357999/selenium-not-detecting-an-element-and-i-cant-switch-to-shadow-root-internal-str/78359878

**Reply by mdmintz** [reply]
The SeleniumBase examples were updated earlier this year to click through Shadow DOM.
